File: BackEnd_Learn/Java/daily_book/src/main/java/com/dsm/daily_book/controller/UserController.java
package com.dsm.daily_book.controller;

import com.dsm.daily_book.dto.UserDTO;
import com.dsm.daily_book.dto.TokenDTO;
import com.dsm.daily_book.service.user.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
public class UserController {
    private final UserService userService;

    @PostMapping("/verify")
    public void mail(@RequestBody UserDTO.mail request){
        userService.sendAuthCode(request);
    }

    @PutMapping("/verify")
    public void verifyCode(@RequestBody UserDTO.authCode request){
        userService.verifyAuthCode(request);
    }

    @PostMapping("/register")
    public void register(@RequestBody UserDTO.register request){
        userService.register(request);
    }

    @PostMapping("/auth")
    public TokenDTO.token auth(@RequestBody UserDTO.auth request){
        return userService.auth(request);
    }

    @PatchMapping("/nickname")
    public void nickname(@RequestBody UserDTO.changeNickname request){
        userService.changeNickname(request);
    }
}

File: BackEnd_Learn/kotlin/entity/src/main/kotlin/com/example/entity/controller/UserController.kt
package com.example.entity.controller

import com.example.entity.dto.UserDTO
import com.example.entity.service.UserService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class UserController (val userService: UserService) {

    @PostMapping("/user/register")
    fun userRegister(@RequestBody request: UserDTO.UserInfo){
        userService.register(request)
    }

    @PostMapping("/user/auth")
    fun userLogin(@RequestBody request: UserDTO.UserInfo){
        userService.login(request)
    }

}